version.gradle.kts

```diff
@@ -1,8 +1,13 @@
+val snapshot = true
+
 allprojects {
+  var v = "1.14.0"
   val release = findProperty("otel.release")
   if (release != null) {
-    version = "1.14.0-" + release + "-SNAPSHOT"
-  } else {
-    version = "1.14.0-SNAPSHOT"
+    v += "-" + release
   }
+  if (snapshot) {
+    v += "-SNAPSHOT"
+  }
+  version = v
```

Comment:
@anuraaga if you know of a more kotlin-y way to write this let me know

Reply:
An extra line but guess this might be more kotlin-y (priority on not using `var`).

```kotlin
val ver = "1.14.0"

version = if (release != null) {
  "${ver}-${release}"
} else if (snapshot) {
  "${ver}-SNAPSHOT"
} else {
  ver
}
```

`ver + if (release != null)` ... is also an option that some may like, but I personally don't

Reply:
there's also 😢 

```
"${ver}-${release}-SNAPSHOT"
```

Reply:
Oh! Then current seems good